src/main/asciidoc/reference/elasticsearch-repository-queries.adoc

```diff
@@ -0,0 +1,157 @@
+[[elasticsearch.query-methods]]
+= Query methods
+
+[[elasticsearch.query-methods.finders]]
+== Query lookup strategies
+
+The Elasticsearch module supports all basic query building feature as String,Abstract,Criteria or have it being derived from the method name.
+
+=== Declared queries
+
+Deriving the query from the method name is not always sufficient and/or may result in unreadable method names. In this case one might make either use of `@Query` annotation (see <<elasticsearch.query-methods.at-query>> ).
+
+[[elasticsearch.query-methods.criterions]]
+== Query creation
+
+Generally the query creation mechanism for Elasticsearch works as described in <<repositories.query-methods>> . Here's a short example of what a Elasticsearch query method translates into:
+
+.Query creation from method names
+====
+[source,java]
+----
+public interface BookRepository extends Repository<Book, String>
+{
```

Comment:
Nit: Let's pull the curly brace one line up.

Reply:
done